platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/util/PropertiesFileSupport.java

```diff
@@ -0,0 +1,100 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+
+/**
+ * Useful for scanning a directory for properties files and aggregating the data into maps and lists.
+ */
+public class PropertiesFileSupport {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PropertiesFileSupport.class);
+
+    private static final String PROPERTIES_EXTENSION = ".properties";
+
+    /**
+     * Deserializes all on-disk properties files in the given directory into a list of maps.
+     *
+     * @param directoryPath the path to the directory of properties files
+     * @return a list of maps holding the values of the properties files in the directory the given
+     * path pointed at, which can be empty if no properties were found.
+     */
+    public List<Map<String, String>> loadPropertiesFilesInDirectory(String directoryPath) {
+        List<Map<String, String>> loadedPropertiesFiles = new ArrayList<>();
+        if (!StringUtils.isEmpty(directoryPath)) {
+            File directory = new File(directoryPath);
+            if (directory.exists()) {
+                File[] propertyFiles = directory.listFiles((dir, name) -> name.endsWith(
+                        PROPERTIES_EXTENSION));
+                if (propertyFiles != null) {
+                    for (File propertyFile : propertyFiles) {
+                        loadedPropertiesFiles.add(loadPropertiesFile(propertyFile));
+                    }
+                } else {
+                    LOGGER.debug("Property load target {} was not a directory", directoryPath);
+                }
+            }
+        }
+        return loadedPropertiesFiles;
+    }
+
+    /**
+     * Deserializes an on-disk properties file into a map.
+     *
+     * @param filePath the path to the file of properties
+     * @return a map of the values in the properties file the provided path pointed at, which can
+     * be empty if no properties were found.
+     */
+    public Map<String, String> loadSinglePropertiesFile(String filePath) {
+        Map<String, String> propertyMap = new HashMap<>();
+        if (!StringUtils.isEmpty(filePath)) {
+            File propertyFile = new File(filePath);
+            if (propertyFile.exists()) {
+                if (!propertyFile.isDirectory()) {
+                    propertyMap.putAll(loadPropertiesFile(propertyFile));
+                } else {
+                    LOGGER.debug("Property load target {} was a directory", filePath);
+                }
+            } else {
+                LOGGER.debug("Property load target {} does not exist", filePath);
+            }
+        }
+        return propertyMap;
+    }
+
+    /**
+     * Helper method for working with the {@link PropertiesLoader}.
+     */
+    private Map<String, String> loadPropertiesFile(File propertiesFile) {
+        Map<String, String> propertyMap = new HashMap<>();
+        if (propertiesFile != null && propertiesFile.exists()) {
```

Comment:
This code was left precisely the same as it was in its prior state, and back then it was working fine. I'm concerned with tweaking this prior to a release. I'll add a note about this to the tech debt ticket.